src/sync.mjs

```diff
@@ -1,37 +1,61 @@
 
-export default function sync(animationName) {
+export default function sync(animationNameOrNames) {
+  const animationNames = new Set(
+    Array.isArray(animationNameOrNames) ? animationNameOrNames : [animationNameOrNames]
+  );
   const elements = new Set();
-  let eventTime;
-  let lastIterationTS = now();
-  const shouldSync = Array.isArray(animationName)
-    ? (event) => animationName.indexOf(event.animationName) > -1
-    : (event) => event.animationName === animationName;
+  let animationDuration;
+  let isPaused = false;
+  let lastInterationTimestamp = 0;
+
+  function shouldSync(event) {
+    return animationNames.has(event.animationName);
+  }
+
+  function validate(el) {
+    const isValid = document.body.contains(el);
+    if (!isValid) {
+      elements.delete(el);
+    }
+    return isValid;
+  }
+
+  function init() {
+    setTimeout(restart, animationDuration);
+  }
+
+  function restart() {
+    this.stop();
+    setTimeout(this.start, 50);
```

Comment:
I think there's a potential bug here: `this` in this case will be `window` if I understand correctly.

Reply:
ah yes - that is not right!

Reply:
Fix published as v0.4.1 - thanks